pebble/src/test/java/io/knotx/te/pebble/PebbleTemplateEngineTest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2019 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.te.pebble;
+
+import static io.knotx.junit5.assertions.KnotxAssertions.assertEqualsIgnoreWhitespace;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+import com.google.common.util.concurrent.UncheckedExecutionException;
+import com.mitchellbosecke.pebble.error.AttributeNotFoundException;
+import com.mitchellbosecke.pebble.error.ParserException;
+import com.mitchellbosecke.pebble.error.RootAttributeNotFoundException;
+import io.knotx.fragments.api.Fragment;
+import io.knotx.junit5.util.FileReader;
+import io.knotx.te.pebble.options.PebbleEngineOptions;
+import io.knotx.te.pebble.options.PebbleEngineSyntaxOptions;
+import io.vertx.core.json.JsonObject;
+import java.io.IOException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+class PebbleTemplateEngineTest {
+
+  private PebbleEngineOptions options;
+
+  @BeforeEach
+  void setUp() {
+    options = new PebbleEngineOptions().setCacheSize(100L);
+  }
+
+  @Test
+  @DisplayName("Expect template with custom syntax to be filled properly")
+  void renderTemplateWithCustomDelimiters() throws IOException {
+    options.setSyntax(getCustomSyntaxOptions());
+    final PebbleTemplateEngine templateEngine = new PebbleTemplateEngine(options);
+    final Fragment fragment = mockFragmentFromFile("templates/serviceCustomSyntax.peb",
+        "data/serviceContext.json");
+    final String result = templateEngine.process(fragment).trim();
+    final String expected = FileReader.readText("results/service").trim();
+    assertEqualsIgnoreWhitespace(expected, result);
+  }
+
+  @Test
+  @DisplayName("Expect exception to be thrown when context is empty and strict mode is enabled")
```

Comment:
Those tree tests below tests similar approaches. Can we convert them to parametrised tests and add cases when strict variables are disabled?